Userland/Libraries/LibJS/Runtime/StringObject.cpp

```diff
@@ -61,11 +59,7 @@ static Optional<PropertyDescriptor> string_get_own_property(GlobalObject& global
     // NOTE: If the property name is a number type (An implementation-defined optimized
     // property key type), it can be treated as a string property that has already been
     // converted successfully into a canonical numeric index.
```

Comment:
I think this comment can be removed now, the optimization is hidden in `canonical_numeric_index_string`

Reply:
Well as @IdanHo said we don't follow the spec exactly since it says:
```
// 3. If Type(P) is String, then
```
which we do with
```cpp
if (property_name.is_string() || property_name.is_number()) {
```
So it is a seeming mismatch but if it is clear enough I'm fine with removing the comments

Reply:
Whoops here this is not actually the case but in TypedArray.h it is

Reply:
no, this is not the case here :)

Here, the spec says `// 4. Let index be ! CanonicalNumericIndexString(P).` and we do `auto index = canonical_numeric_index_string(global_object, property_name);` - no additional comment needed.

`if (property_name.is_string() || property_name.is_number()) {` is elsewhere and there we should keep the comment.

Reply:
Removed the note here and kept the top notes in TypedArray